API/Controllers/UsersController.cs

```diff
@@ -0,0 +1,149 @@
+using System.Diagnostics.CodeAnalysis;
+using API.DTOs;
+using API.Enums;
+using API.Services.Interfaces;
+using API.Utilities;
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[ApiController]
+[ApiVersion(1)]
+[Authorize(Roles = "system, admin")] // Internal access only
+[Route("api/v{version:apiVersion}/[controller]")]
+[SuppressMessage("ReSharper", "RouteTemplates.ActionRoutePrefixCanBeExtractedToControllerRoute")]
+public class UsersController(IUserService userService, IOAuthClientService clientService) : Controller
+{
+    private readonly IUserService _userService = userService;
+    private readonly IOAuthClientService _clientService = clientService;
```

Comment:
Given the variables are made accessible to the entire class with this new constructor syntax, what do you think of removing the `private readonly` fields in classes going forward? I've been doing this in my new classes.

Reply:
That's fine with me. I've been inconsistently using either or, so let's go with using primary constructors where we can then.